extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js

```diff
@@ -0,0 +1,375 @@
+					// nothing in cache, just fetch
+					this.updateWikiFetch(wikiId);
+					return;
+				}
+				var data = this.notificationsCache[wikiId],
+					wikiCount = parseInt(this.$wallNotifications.find(
+						'li[data-wiki-id=' + wikiId + ']'
+					).data('unread-count'), 10);
+				// no change in notifications, update from cache
+				if (wikiCount === data['unread']) {
+					this.updateWikiHtml(wikiId, data);
+
+					// different number of unread, fetch from server
+					// in the meantime, show old data
+				} else {
+					this.updateWikiHtml(wikiId, data);
+					this.updateWikiFetch(wikiId);
+				}
+			},
+
+			updateWikiFetch: function (wikiId) {
+				var isCrossWiki = (wikiId === this.cityId) ? '0' : '1',
+					data = {
+						wikiId: wikiId,
+						isCrossWiki: isCrossWiki
+					};
+
+				$.extend(data, this.getUrlParams());
+
+				nirvana.sendRequest({
+					controller: 'WallNotificationsExternalController',
+					method: 'getUpdateWiki',
+					type: 'GET',
+					data: data,
+					callback: this.proxy(function (data) {
+						if (data.status !== true || data.html === '') {
+							return;
+						}
+						this.updateWikiHtml(wikiId, data);
+						this.notificationsCache[wikiId] = data;
+					}),
+					onErrorCallback: this.proxy(function (jqXHR, textStatus, errorThrown) {
+						if (jqXHR !== undefined && jqXHR.status !== undefined && jqXHR.status === 501) {
+							var data = {};
+							data.html = '<li class="notifications-empty">' + $.msg('wall-notifications-wall-disabled') + '</li>';
+							this.updateWikiHtml(wikiId, data);
+							this.notificationsCache[wikiId] = data;
+						}
+					})
+				});
+			},
+
+			updateWikiHtml: function (wikiId, data) {
+				var wikiEl = this.$wallNotifications.find('li[data-wiki-id=' + wikiId + ']'),
+					wikiLi = wikiEl.find('.notifications-for-wiki-list');
+
+				wikiLi.html(data.html);
+
+				if (this.wikiShown[wikiId]) {
+					wikiEl.addClass('show');
+				}
+
+				wikiLi.find('time').timeago();
+
+				// hijack links for other wikis - open them in new window
+				if (wikiId !== this.currentWikiId) {
+					wikiEl.find('a').attr('target', '_blank');
+
+					// temporary fix, should be changed on server side
+					wikiEl.find('.read').remove();
+				}
+
+				this.setNotificationsHeight();
+			},
+
+			getLastSeenCount: function () {
+				return $.cookies.get('WallUnreadCount') || 0;
+			},
+
+			setLastSeenCount: function (val) {
+				$.cookies.set('WallUnreadCount', val);
+			},
+
+			proxy: function (func) {
+				return $.proxy(func, this);
+			},
+
+			getUrlParams: function () {
+				var data = {},
+					qs = Wikia.Querystring(),
+					lang, skin;
+
+				skin = qs.getVal('useskin');
+				if (skin) {
+					data.useskin = skin;
+				}
+
+				lang = qs.getVal('uselang');
+				if (lang) {
+					data.uselang = lang;
+				}
+
+				return data;
+			},
+
+			setNotificationsHeight: function () {
+				var isDropdownOpen = this.$wallNotifications.hasClass('show'),
+					height = 0,
+					msgHeight = 0;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "msgHeight" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)